Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiGpuProfiler.h

```diff
@@ -250,6 +253,42 @@ namespace AZ
 
         };
 
+        class ImGuiGpuMemoryView
+        {
+        public:
+            // Draw the overall GPU memory profiling window.
+            void DrawGpuMemoryWindow(bool& draw);
+
+            // Draw the heap usage pie chart
+            void DrawPieChart(const AZ::RHI::MemoryStatistics::Heap& heap);
+
+            // Update the saved pointers in m_tableRows according to new data/filters
+            void UpdateTableRows();
+
+            void DrawTable();
+
+            // Sort the table according to the appropriate column.
+            void SortTable(ImGuiTableSortSpecs* sortSpecs);
```

Comment:
Is there a use-case for DrawPieChart() and SortTable() to be public? Seems like these can be encapsulated.

Reply:
You're right, made everything private except for DrawGpuMemoryWindow since that's the main entrypoint that should be called.